### PR TITLE
Further enhanced the syntax support

### PR DESCRIPTION
**Pull request by mclaughlinconnor**
Force pushed to #3 and now I can't reopen

---

I realise this is a pretty hefty PR, but I'm not sure what your thoughts are on this repo, or if you're planning on keeping in maintained.

I've added syntax support for nearly everything (I think anyway) -- there are a few small problems with some of the JS style attributes, etc., but I hope to have them sorted soon-ish.

If you'd rather, though (and I'm guessing this will be the case seeing as I've rewritten most of it, and to avoid any upstreaming requests), I can just "unfork" my repo to be maintained separately by me.

Let me know what your thoughts are :)

### CODE REVIEW

**Comment by JulesNP** [comment]
I don't think this is being maintained by @zealot128 anymore, as I've had a pull request pending since May 2022 with no response. Your best bet is to put in a pull request with the nvim-treesitter project itself to switch to your fork.

**Comment by zealot128** [comment]
@JulesNP @mclaughlinconnor ups, I didn't noticed this PR.  The default notification was only on for Mentions/watched threads for both of your PR's. (Maybe because this is a fork itself...?)

Will check out if it works for "my" pug files, and if yes, I see no issues to give you commit rights. I only forked it myself to make it work for my use cases anyway. It would be best for this repo to be part of @pugjs group, but I am not in contact with anyone from that team.

**Comment by zealot128** [comment]
On cursory glance it parses most of the vue-pug files that I took a look on. Will use the fork for a couple of workdays.
Maybe @JulesNP can also check out, if this PR fixes the quotation issue, because I think that merging both will be hard because of conflicts.

**Comment by mclaughlinconnor** [comment]
@JulesNP No, my code doesn't fix your string quoting issue. I probably could have, but I didn't want to shamelessly steal your code :)

I haven't tested the code, but I know for sure it will have conflicts.

I think you'll also want to use the same code for `quoted_javascript` nodes

**Comment by mclaughlinconnor** [comment]
Re living under @pugjs, I agree, but there doesn't seem to be much activity in the group. I'm not in contact with anyone either, but @ForbesLindesay is pretty active on Github, so maybe we'll get a response with his thoughts on moving this into the group.

**Comment by JulesNP** [comment]
> On cursory glance it parses most of the vue-pug files that I took a look on. Will use the fork for a couple of workdays. Maybe @JulesNP can also check out, if this PR fixes the quotation issue, because I think that merging both will be hard because of conflicts.

It doesn't fix my quoting issue, but everything else with this version is working well. I'll withdraw my pull request and resubmit an updated version once @mclaughlinconnor's changes are merged.

File: README.md
Comment (zealot128) [inline README.md]: Can you word this differently? It's not for either Vue or Angular only now :)

**Review by zealot128 — CHANGES_REQUESTED** [review]

**Comment by zealot128** [comment]
Haven't noticed any regression in the highlighting of my Vue/pug files.
Great work!!

Only some wording in the README is Angular only. Can you adjust that please? Now the grammar is not only useful for us Vue or Angular folks, but almost general purpose.

File: README.md
Comment (mclaughlinconnor) [reply README.md]: Thanks @zealot128 I've update the  README to be more inclusive :+1:

**Comment by zealot128** [comment]
Awesome! Thanks for the huge contribution.

**Pull request merged**
